api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -258,6 +270,23 @@ public <T> String predicate(UnboundPredicate<T> pred) {
     }
   }
 
+  private static <T> List<String> abbreviateValues(List<String> sanitizedValues) {
+    if (sanitizedValues.size() >= LONG_IN_PREDICATE_ABBREVIATION_THRESHOLD) {
+      Set<String> distinctValues = ImmutableSet.copyOf(sanitizedValues);
+      if (distinctValues.size()
+          <= sanitizedValues.size() - LONG_IN_PREDICATE_ABBREVIATION_MIN_GAIN) {
```

Comment:
This logic of de-duplicating sanitized values works well only when we're dealing with integers because sanitized integers are represented by a limited set of values like `2-digit-int`, `3-digit-int` etc.
But when we have string values which are sanitized to hashes (e.g. `(hash-1b409883), (hash-53cd6d46), (hash-24add70a), (hash-7df3cf93), ...` this logic isn't helpful in abbrevaiting long IN list.
Since we're already dealing with sanitized values, I'm wondering why we're making the effort to always print all the distinct sanitized values ? 
Can we make this simpler and just print first 2 and last distinct sanitized values when the number of distinctValues exceeds some threshold ?

Reply:
> Can we make this simpler and just print first 2 and last distinct sanitized values when the number of distinctValues exceeds some threshold ?

yes! (or a few more)